### ISSUE TITLE
There is no formatter for 'json' files installed.

### ISSUE THREAD

**Issue by ssbarnea** [microsoft/vscode#140268]
When trying to reformat a JSON file using `F1 > Format`, I get a confusing dialog that reports that no formatter for json is installed. When pressing to install one I get redirected to marketplace listing of JSON extensions, where there are thousands of them and after trying 3 of them that did not sort the issue I decided to file a bug.

I think that vscode is expected to be able to format JSON without additional extensions and I am also aware of one or two similar bugs which were marked as fixed about 2 years ago.

That must be something new, especially as I used extension bisect feature to test if another extension is affecting it and the result was negative, reproduced without extensions.

```
Version: 1.63.2
Commit: 899d46d82c4c95423fb7e10e68eba52050e30ba3
Date: 2021-12-15T09:37:28.172Z (3 wks ago)
Electron: 13.5.2
Chromium: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin x64 21.2.0
```

**Comment by gjsjohnmurray** [comment]
What is your `json.format.enable` setting?

In the Extensions view does the search for `@builtin json language features` show the built-in extension? Is it enabled?

**Comment by ssbarnea** [comment]
One interesting thing that i observed, the "JSON Language Features" appears as disabled but I really doubt I ever disable a build-it. The basics one is enable.
![](https://sbarnea.com/ss/Screen-Shot-2022-01-07-11-36-08.66.png)

`json.format.enable` is not present in config.

Another interesting thing is that I observed that while "Format" does not work, the "Format (forced)" seems to work.
![](https://sbarnea.com/ss/Screen-Shot-2022-01-07-11-38-21.16.png)

After enabling "JSON Language Features" behaviour changed and now I see a different dialog when using Format:
![](https://sbarnea.com/ss/Screen-Shot-2022-01-07-11-39-23.80.png)

The funny bit is that the "Configure" option is useless as it does not allow me to configure/select a particular formatter. In fact I have no clue which are those formatters and the message does not mention them. I can guess that one of them is provided by "JSON Language Features"  built-in but who is the other one? I think it would be very useful to make this dialog more explicit as it can easily become a needle-in the haystack search experience.

**Comment by aeschli** [comment]
The interesting question is, how the JSON language extension got disabled.

Can you file a separate issue on improving the formatter selection dialog?
This is how the formatter selection dialog looks for me:
![image](https://user-images.githubusercontent.com/6461412/148551726-dd68587c-a202-466c-acda-9d83c3f92d48.png)

For historical reasons. actual formatters don't have a name so we show the name of the extension that contributes it.
FYI @jrieken Maybe you already have an issue to improve the labels, e.g. allow formatters to provide a name as well.

**Comment by ssbarnea** [comment]
I think I may have found the culprit, the dialog looks the same for me.

At least in one case I ended up with 100% CPU usage when I used Format, but I was not able to reproduce it. I can assume it was from Prettier extension but I am not 100% sure.

I guess that once you pick a formatter an entry similar to the one below is added to user `settings.json` (or json instead of jsonc based on file):
```
    "[jsonc]": {
        "editor.defaultFormatter": "esbenp.prettier-vscode"
    },
```
